### PR TITLE
Drop @expo/config-plugins dependency in favor of an expo peer dependency

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

We are having issues of users having conflicting versions of `@expo/config-plugins`, this isn't a huge issue with the Expo SDK since everything is versioned but it seems fine to just depend on the version of config-plugins that is in the SDK.

# Test Plan

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by tsapeta — APPROVED** [review]

File: packages/expo-apple-authentication/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,8 @@
 
 ### 💡 Others
 
+- Drop `@expo/config-plugins` dependency in favor of peer dependency on `expo`.
```
Comment (EvanBacon) [inline packages/expo-apple-authentication/CHANGELOG.md:13]:
```suggestion
- Drop `@expo/config-plugins` dependency in favor of peer dependency on `expo`. ([#18595](https://github.com/expo/expo/pull/18595) by [@EvanBacon](https://github.com/EvanBacon))
```

**Comment by brentvatne** [comment]
this is great! i missed this pr yesterday and discussed the same thing in #sdk-release after you'd already opened this 😂 

- should we do this for `@expo/config` and `@expo/config-types` as well?
- what do you think about re-exporting config-plugins and other config packages from the expo package in the same way that we do with metro-config?

**Review by brentvatne — APPROVED** [review]

**Comment by EvanBacon** [comment]
@brentvatne Probably fine for config types, config shouldn’t really be a big issue since it hardly ever changes. 
I think maybe we should play the re-exports by ear and wait until we have a proper use case for them.

**Comment by brentvatne** [comment]
@EvanBacon - the concern is that @expo/config also depends on @expo/config-plugins

**Comment by KiwiKilian** [comment]
Should the dependency also be remove in community-driven plugins like [@rnmapbox/maps](https://github.com/rnmapbox/maps)?

**Pull request merged**